src/Compiler/Lexer.cpp

```diff
@@ -44,7 +44,7 @@ namespace Ark::internal
             if (current == '\n')
             {
                 line++;
-                character = 0;
+                character = -1; // before first character
```

Comment:
Alas this is an undefined behaviour since `character` is of type `std::size_t` (typedef of `unsigned long int` on x64), even though that kind of work in the wild. I would suggest changing its type to a signed one, like `long`

Reply:
I shifted this newline handling code to the end of the loop. So it resets the line and character count when encountering a newline and increments the character count otherwise.